src/main/java/tests/CommentPostTests.java

```diff
@@ -65,67 +61,76 @@ public void setup() {
+        assert(commentPostRepository.findByContext(post).get(0).getText().equals(commentText));
     }
 
     @Test
     public void testAddCommentToInvalidSubmission() throws Exception {
+        String commentText = "test comment on invalid submission";
         mvc
                 .perform(post("/posts/invalid")
                         .with(csrf())
                         .with(user("sarran"))
-                        .param("text", "This is a test comment"))
+                        .param("text", commentText))
                 .andExpect(status().isOk())
                 .andExpect(view().name("pageNotFound"));
     }
 
     @Test
     public void testAddInvalidCommentToValidSubmission() throws Exception {
         SubmissionPost post = new SubmissionPost();
-        post.setTitle("This is my AMA");
+        post.setTitle("Add invalid comment to this submission");
         post.setText("Test text.");
-        submissionPostRepository.save(post);
+        post = submissionPostRepository.save(post);
         mvc
                 .perform(post("/posts/" + post.getTitle())
                         .with(csrf())
                         .with(user("sarran"))
                         .param("text", ""))
                 .andExpect(status().isOk())
-                .andDo(print())
                 .andExpect(view().name("pageNotFound"));
+        assert(commentPostRepository.findByContext(post).size() == 0);
     }
 
     @Test
     public void testReplyToValidComment() throws Exception {
         CommentPost comment = new CommentPost();
         comment.setText("this is a base comment.");
         comment = commentPostRepository.save(comment);
+
+        String replyText = "This is a reply.";
         mvc
                 .perform(post("/comments/" + comment.getId().toString() + "/reply")
                         .with(csrf())
                         .with(user("sarran"))
-                        .param("reply", "this is a reply"))
-                .andDo(print())
+                        .param("reply", replyText))
                 .andExpect(status().is3xxRedirection());
+        assert(commentPostRepository.findByContext(comment).get(0).getText().equals(replyText));
     }
 
     @Test
-    public void testReplyToInValidComment() throws Exception {
+    public void testReplyToInvalidComment() throws Exception {
+
+        String replyText = "This is a reply to an invalid comment.";
         mvc
-                .perform(post("/comments/2/reply")
+                .perform(post("/comments/10/reply")
                         .with(csrf())
                         .with(user("sarran"))
-                        .param("reply", "this is a reply"))
-                .andExpect(status().isOk())
-                .andExpect(view().name("pageNotFound"));
+                        .param("reply", replyText))
+                .andExpect(status().is3xxRedirection());
+
+        assert(commentPostRepository.findByText(replyText).size() == 0);
```

Comment:
I would get rid of `findByText()` and just use `findAll().get(0)` since you know that there will only be one comment in that repository. `findByText()` is a fairly brittle way of searching the repo since they aren't unique, and I don't think that it's a method we should have in our production code